aspnetcore/fundamentals/servers/kestrel/endpoints.md

```diff
@@ -314,7 +321,59 @@ webBuilder.ConfigureKestrel(serverOptions =>
 });
 ```
 
-SNI support requires:
+### SNI in configuration
+
+Kestrel supports SNI defined in configuration. An endpoint can be configured with an `Sni` object that contains a mapping between server names and HTTPS options. The following configuration adds an endpoint named `MySniEndpointName` that uses SNI to select HTTPS options based on the server name:
+
+```json
+{
+  "Kestrel": {
+    "Endpoints": {
+      "MySniEndpointName": {
+        "Url": "https://*",
+        "Sni": {
+          "a.example.org": {
+            "Protocols": "Http1AndHttp2",
+            "SslProtocols": [ "Tls11", "Tls12", "Tls13"],
+            "Certificate": {
+              "Subject": "<subject; required>",
+              "Store": "<certificate store; required>",
+            },
+            "ClientCertificateMode" : "NoCertificate"
```

Comment:
The code-first approach relies on API docs to explain what parameters and values can be used. Do we need to provide more explicit docs here for the config approach so people will understand the parameters and values? The sample is pretty good, but it's not clear if it's exhaustive. Or are these all documented elsewhere that we can link to?

Certs: #certificate-sources
Protocols: #set-the-http-protocol-from-configuration
SslProtocols: https://docs.microsoft.com/en-us/dotnet/api/system.security.authentication.sslprotocols?view=net-5.0 (and expand #ssltls-protocols)
ClientCertificateMode: https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.server.kestrel.https.clientcertificatemode?view=aspnetcore-5.0

Reply:
Good point. This is exhaustive, but it's not clear that's the case without saying so explicitely somewhere.